### PR TITLE
[Request][8.11-8.13] Doc privs required to create and manage the `.lists` data stream

### PR DESCRIPTION
**Pull request by nastasha-solomon**
Contributes to https://github.com/elastic/security-docs/issues/4477

Previews:
- [Lists API](https://security-docs_bk_4524.docs-preview.app.elstc.co/guide/en/security/master/lists-api-overview.html#_kibana_role_requirements_3) - Added a new section titled "Lists requirements" to describe the pre-reqs and required privs for working with lists. 
- [Exceptions API](https://security-docs_bk_4524.docs-preview.app.elstc.co/guide/en/security/master/exceptions-api-overview.html): Added a new section titled "Exception requirements" to describe the pre-reqs and required privs for working with exceptions using value lists. 
- [Detections prerequisites and requirements | Enable and access detections](https://security-docs_bk_4524.docs-preview.app.elstc.co/guide/en/security/master/detections-permissions-section.html#enable-detections-ui) - Added a new role that detailed the privs needed to create the `.lists` and `.items` data streams.

Note: There is no Serverless PR because our Serverless docs don’t contain API docs and our detection reqs [topic](https://docs.elastic.co/serverless/security/detections-requirements) doesn't expose the privs table atm.

### CODE REVIEW

File: docs/detections/api/lists/lists-api-overview.asciidoc [hunk 1]
```diff
@@ -60,13 +60,10 @@ data streams for the {kib} space (see <<lists-index-api-overview>>).
 [float]
 === Kibana role requirements
 
-To create list containers and items, the user role for the {kib} space must
-have:
+To turn on the lists feature, a user with the `manage` cluster privilege must visit the Rules page in your Kibana space.
```
Comment (vitaliidm) [inline docs/detections/api/lists/lists-api-overview.asciidoc:63]: without .lists,.items privileges feature won't be 'turn on', just `.manage` cluster is not enough

Comment (vitaliidm) [reply docs/detections/api/lists/lists-api-overview.asciidoc:63]: Maybe we can rephrase it as:

To turn on the lists feature, user with the following privileges must visit the Rules page in your Kibana space:

- `manage` cluster privilege
- `read` and `write` index privileges for the `.lists` and `.items` data streams (the system data stream used for storing exception lists).
- {kib} space `All` privileges for the `Security` and `Saved Objects Management`
features (see {kibana-ref}/xpack-spaces.html#spaces-control-user-access[Feature access based on user privileges]).

Comment (nastasha-solomon) [reply docs/detections/api/lists/lists-api-overview.asciidoc:63]: OK, let me try to summarize what I've gathered so far. It appears there's two use cases that need to be doc'd.

**Use case 1:** Create the `.lists` data stream, which can contain lists and lists items. 

To create the `.lists` data stream, a user with the following privs must visit the Rules page in your Kibana space:

- `manage` cluster privilege
- `read` and `write` index privileges for the `.lists` and `.items` data streams (the system data stream used for storing exception lists).
- {kib} space `All` privileges for the Security and Saved Objects Management features (see {kibana-ref}/xpack-spaces.html#spaces-control-user-access[Feature access based on user privileges]).

**Use case 2:** Manage lists 

To create, edit, and delete lists, a user needs the following privs:

- `read` and `write` index privileges for the `.lists` and `.items` data streams (the system data stream used for storing exception lists).
- {kib} space `All` privileges for the Security and Saved Objects Management features (see {kibana-ref}/xpack-spaces.html#spaces-control-user-access[Feature access based on user privileges]).

Is this correct? Also, do users need different privileges to manage list items? Or are the required privs the same as what they'd need to manage lists?

Comment (nastasha-solomon) [reply docs/detections/api/lists/lists-api-overview.asciidoc:63]: Gotcha and thanks @vitaliidm for providing an example to illustrate the relationship between the  `.lists` and `.items` data streams and the objects they contain. Your explanation and examples will actually help address some of the asks in two other open doc issues (https://github.com/elastic/security-docs/issues/3754 and https://github.com/elastic/security-docs/issues/4116), so they're providing value past this one doc task. 

Moving onto summarizing this thread, here's my understanding of what we've discussed so far:
- Creating the `.lists` and `.items` data streams are pre-requisites to creating and managing lists. To create these data streams, a user with the following privs needs to visit the Rules page:
   - `manage` cluster privilege
   - `read` and `write` index privileges for the `.lists` and `.items` data streams 
   - Kibana space `All` privileges for the Security and Saved Objects Management features 
- Once the data streams have been created, users only need the following privs for managing (create/read/edit/delete) **lists** and **list items**:
   - `read` and `write` index privileges for the `.lists` and `.items` data streams 
   - Kibana space `All` privileges for the Security and Saved Objects Management features 
- Value lists are comprised of two parts: lists and list items. Additional notes on the design:
   - Lists are indexed in the `.lists` data stream and list items are indexed in the `.items` data stream. 
   - List items are indexed as separate documents, and each list item references the list that it's a part of. 

If the above is correct, I'll be make the following changes to this PR:
- [Lists API | Kibana role requirements](https://security-docs_bk_4524.docs-preview.app.elstc.co/guide/en/security/master/lists-api-overview.html#_kibana_role_requirements_3) - Will be making two changes:
    - Explain that the `.lists` and `.items` data streams are pre-requisites to creating and managing lists and describe the required privs to do so. 
    - Will also be changing [the existing text](https://github.com/elastic/security-docs/pull/4524/files#diff-42a39c514ebfe335f624a2e2af4c5bdc43cf4ab5266cf35a597763c0e76619d1R65) from "to create list containers and items..." to "to create and manage list contains and items..."
- [Detections prerequisites and requirements | Enable and access detections](https://security-docs_bk_4524.docs-preview.app.elstc.co/guide/en/security/master/detections-permissions-section.html#enable-detections-ui) - Making a couple changes to the rows I added:
   - Will change [this description](https://github.com/elastic/security-docs/pull/4524/files#diff-7071d3f8f16c85ba42ead73e4dfde38998099dc9cba07ef323600d4da086ff20R129) to "Create data streams for value lists" or "Create data streams for lists". Let me know if these work or if you think we need to go in a different direction. 
   - Will update the required privs for creating the `.lists` and `.items` data streams.

Comment (vitaliidm) [reply docs/detections/api/lists/lists-api-overview.asciidoc:63]: I think `view_index_metadata` is missing in list of privs for `.lists` and `.items`.
 This privilege mentioned on https://www.elastic.co/guide/en/security/current/detections-permissions-section.html#enable-detections-ui

Comment (nastasha-solomon) [reply docs/detections/api/lists/lists-api-overview.asciidoc:63]: Hey, @vitaliidm. [Here](https://docs.google.com/spreadsheets/d/1COlkql3q4Py4NVraBMXo4MRiljD2W-17mADfemLAV-0/edit?usp=sharing) are my testing notes. I was able to get in a lot of good testing from the UI, but was running into 403 errors when testing the list APIs. 

The TLDR of it is: 
- Users need the following privs to create the `.lists` and `.items` data streams:        
   - `manage` cluster privilege
   - `read`, `write`, `view_index_metadata`, and `manage` index privileges for the `.lists` and `.items` data streams 
   - Kibana space `All` privileges for the Security and Saved Objects Management features 
- Once the data streams are created, users need everything above, except for the `manage` cluster priv.

Comment (nastasha-solomon) [reply docs/detections/api/lists/lists-api-overview.asciidoc]: Quick update on the API testing: 
I was able to figure out why I was why I was getting the 403 errors. I was testing the lists feature in a secondary Kibana space (i.e., a non-default Kibana space) and needed to specify the space name in the request URL.

File: docs/detections/api/exceptions/exceptions-api-overview.asciidoc [hunk 2]
```diff
@@ -37,19 +37,13 @@ entities:
 
 image::images/exceptions-logic.png[]
 
-IMPORTANT: Before you can create exceptions, you must create `.lists` and
+[float]
+=== Exceptions prerequisites
+
+Before you can create exceptions, you must create `.lists` and
```
Comment (vitaliidm) [inline docs/detections/api/exceptions/exceptions-api-overview.asciidoc:43]: is this necessary? It's possible to create exception without using lists.

Just tested for user on space which does not have lists data streams.

Maybe we should add note

> Before you can create exceptions that uses lists, you must create `.lists` and

File: docs/detections/api/lists/lists-api-overview.asciidoc [hunk 3]
```diff
@@ -54,19 +54,12 @@ Use an <<exceptions-api-create-exception-item, exception item>> to define the
...
-* `read` and `write` index privileges for the
-`.lists` and `.items` data streams (the system data stream used for storing exception lists).
-* {kib} space `All` privileges for the `Security` and `Saved Objects Management`
-features (see
-{kibana-ref}/xpack-spaces.html#spaces-control-user-access[Feature access based on user privileges]).
+[float]
+=== Required privileges 
 
-See <<detections-permissions-section>> for a complete list of requirements.
+To create list containers and list items, your role requires privileges to manage lists. Refer to <<enable-detections-ui>> for more information.
```
Comment (vitaliidm) [inline docs/detections/api/lists/lists-api-overview.asciidoc:65]: great to have it as reference, so all relevant info would be stored in one place 👍

File: docs/detections/api/exceptions/exceptions-api-overview.asciidoc (hunk 2 above)
Comment (nastasha-solomon) [reply docs/detections/api/exceptions/exceptions-api-overview.asciidoc:43]: I'm not sure if the .lists` and `.items` data streams are requirements for creating exceptions via the exceptions API. I was mainly reformatting the [information](https://www.elastic.co/guide/en/security/master/exceptions-api-overview.html#_kibana_role_requirements_2) that's currently in the docs.

RE your suggestion:
Are exceptions that use lists essentially [shared exception lists](https://www.elastic.co/guide/en/security/master/shared-exception-lists.html)? If so, I can update the language to reflect that.

Comment (vitaliidm) [reply docs/detections/api/exceptions/exceptions-api-overview.asciidoc:43]: > Are exceptions that use lists essentially [shared exception lists](https://www.elastic.co/guide/en/security/master/shared-exception-lists.html)? If so, I can update the language to reflect that.

No, they are just sharable exception lists. As per https://www.elastic.co/guide/en/security/master/shared-exception-lists.html#add-exception-items, 3.b.

> Operator: Select an operator to define the condition:
    is | is not — Must be an exact match of the defined value.
    is one of | is not one of — Matches any of the defined values.
    exists | does not exist — The field exists.
    is in list | is not in list — Matches values in a value list.

Comment (nastasha-solomon) [reply docs/detections/api/exceptions/exceptions-api-overview.asciidoc:43]: Gotcha, ok. So if I'm understanding correctly, there are _no list-related privs required for creating exceptions via the exceptions API_. However, if users want to [create shareable exception lists](https://www.elastic.co/guide/en/security/master/exceptions-api-create-shared-exception-list.html), they need to create the `.lists` and `.items` data streams and require privs to create those data streams and manage lists.

If the above is true, would it make sense to [this information](https://www.elastic.co/guide/en/security/master/exceptions-api-overview.html#_kibana_role_requirements_2) from the exceptions overview page and move it to the [page](https://www.elastic.co/guide/en/security/master/exceptions-api-create-shared-exception-list.html) that describes how to create shared exception lists via the API? Are are there any other places in the exception API docs that require this information?

Comment (vitaliidm) [reply docs/detections/api/exceptions/exceptions-api-overview.asciidoc:43]: I don't think .lists and .items privs required to create shared exceptions lists or exception item or exceptions, **unless**
operator
> is in list | is not in list — Matches values in a value list.
need to be used  in it.

For simplicity, we may leave that both of these data streams needed, but in fact they are not required.

Comment (nastasha-solomon) [reply docs/detections/api/exceptions/exceptions-api-overview.asciidoc:43]: Sorry for the late reply, @vitaliidm and thanks for clarifying. I think I understand now. If users are creating regular exceptions or shared exceptions that use value lists, they need to first create the `.lists` and `.items` data streams, and then they need role access to these data streams.

Comment (nastasha-solomon) [reply docs/detections/api/exceptions/exceptions-api-overview.asciidoc]: Addressed this in [699b2a4](https://github.com/elastic/security-docs/pull/4524/commits/699b2a476a6536fb845ce5658467d350b76a5a8f).

File: docs/getting-started/detections-req.asciidoc [hunk 4]
```diff
@@ -130,6 +126,17 @@ a|The `maintenance`, `write`,`read`, and `view_index_metadata` index privileges
 |{kib} space `Read` privileges for the `Security` feature (refer to
 {kibana-ref}/xpack-spaces.html#spaces-control-user-access[Feature access based on user privileges])
 
+|Create the `.lists` and `.items` data streams in your Kibana space
```
Comment (vitaliidm) [inline docs/getting-started/detections-req.asciidoc:129]:
```suggestion
|Create the `.lists` and `.items` data streams in your Kibana space, which are used as lists and items containers
```

**Review by vitaliidm — APPROVED** [review]

File: docs/detections/api/exceptions/exceptions-api-overview.asciidoc [hunk 5]
```diff
@@ -37,19 +37,9 @@ entities:
...
-`.lists` and `.items` data streams (the system data stream used for storing exception lists).
-* {kib} space `All` privileges for the `Security` and `Saved Objects Management`
-features (see
-{kibana-ref}/xpack-spaces.html#spaces-control-user-access[Feature access based on user privileges]).
+=== Exceptions requirements 
 
-See <<detections-permissions-section>> for a complete list of requirements.
+Before you start working with exceptions that use value lists, you must create the `.lists` and `.items` data streams for the relevant {kib} space. To learn how to do this, go to <<lists-index-api-overview>>. 
+ 
+Once these data streams are created, your role need privileges to manage rules. Refer to <<enable-detections-ui>> for a complete list of requirements.
```
Comment (joepeeples) [inline docs/detections/api/exceptions/exceptions-api-overview.asciidoc:45]:
```suggestion
Once these data streams are created, your role needs privileges to manage rules. Refer to <<enable-detections-ui>> for a complete list of requirements.
```

File: docs/detections/api/exceptions/lists-index-api-overview.asciidoc [hunk 6]
```diff
@@ -4,9 +4,10 @@
 Before using exceptions and lists, use the `index` endpoint to create `.lists`
 and `.items` system data streams in the relevant
 {kibana-ref}/xpack-spaces.html[{kib} space].
+//Need to follow up on this
```
Comment (joepeeples) [inline docs/detections/api/exceptions/lists-index-api-overview.asciidoc:7]: Do you still need this? Maybe create a separate ticket for any follow-up work.

File: docs/detections/api/lists/lists-api-overview.asciidoc [hunk 7]
```diff
@@ -54,19 +54,9 @@ Use an <<exceptions-api-create-exception-item, exception item>> to define the
...
-`.lists` and `.items` data streams (the system data stream used for storing exception lists).
-* {kib} space `All` privileges for the `Security` and `Saved Objects Management`
-features (see
-{kibana-ref}/xpack-spaces.html#spaces-control-user-access[Feature access based on user privileges]).
+=== Lists requirements 
 
-See <<detections-permissions-section>> for a complete list of requirements.
+Before you can start using lists, you must create the `.lists` and `.items` data streams for the relevant {kib} space. To learn how to do this, go to <<lists-index-api-overview>>. 
+ 
+Once these data streams are created, your role need privileges to manage rules. Refer to <<enable-detections-ui>> for a complete list of requirements.
```
Comment (joepeeples) [inline docs/detections/api/lists/lists-api-overview.asciidoc:62]:
```suggestion
Once these data streams are created, your role needs privileges to manage rules. Refer to <<enable-detections-ui>> for a complete list of requirements.
```

File: docs/getting-started/detections-req.asciidoc [hunk 8]
```diff
@@ -130,6 +126,17 @@ a|The `maintenance`, `write`,`read`, and `view_index_metadata` index privileges
...
+
+|The `manage` privilege
+a| The `manage`, `write`,`read`, and `view_index_metadata` index privileges for the following data streams, where `<space-id>` is the {kib} space name:
+
+* `.lists-<space-id>`
+* `.items-<space-id>`
+
+|{kib} space `All` privileges for the `Security` and `Saved Objects Management`
+features (refer to {kibana-ref}/xpack-spaces.html#spaces-control-user-access[Feature access based on user privileges]).
```
Comment (joepeeples) [inline docs/getting-started/detections-req.asciidoc:138]: The other rows in this column don't use ending punctuation:
```suggestion
features (refer to {kibana-ref}/xpack-spaces.html#spaces-control-user-access[Feature access based on user privileges])
```

File: docs/detections/api/exceptions/lists-index-api-overview.asciidoc (hunk 6 above)
Comment (nastasha-solomon) [reply docs/detections/api/exceptions/lists-index-api-overview.asciidoc:7]: Whoops, don't need this anymore.

```suggestion
```

**Review by joepeeples — DISMISSED** [review]

File: docs/getting-started/detections-req.asciidoc (hunk 4 above)
Comment (nastasha-solomon) [inline docs/getting-started/detections-req.asciidoc:129]:
```suggestion
|Create the `.lists` and `.items` data streams in your Kibana space

**NOTE**: To initiate the process that creates the `.lists` and `.items` data streams, you must visit the Rules page for each appropriate Kibana space.
```

**Review by vitaliidm — APPROVED** [review]

**Review by natasha-moore-elastic — APPROVED** [review]

**Pull request merged**